.github/workflows/linters.yml

```diff
@@ -0,0 +1,53 @@
+name: Linters
+
+on:
+  pull_request:
+  # Build when a pull request targets main
+    branches:
+      - main
+
+jobs:
+  linter:
+    name: Lint checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout the repository
+        with:
+          fetch-depth: 100
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Check example keywords
+        run: python3 doc/example-keywords.py compare
+      - name: Get changed C++ files
+        if: success() || failure()
+        id: changed-cxx-files
+        uses: tj-actions/changed-files@v37
+        with:
+          files: |
+            **.h
+            **.cpp
+      # Comment blocks starting with "/*!" do not support autobrief
+      - name: Check for problematic Doxygen comment blocks
+        if: (success() || failure()) && steps.changed-cxx-files.outputs.any_changed == 'true'
+        run: |
+          RETCODE=0
+          for file in ${{ steps.changed-cxx-files.outputs.all_changed_files }}; do
+            RESULT=`perl -0777 -ne 'while(m/\n\n *\/\*!.*?\*\//gs){print "$&\n";}' $file`
+            if [ "$RESULT" ]; then
+              echo ------ ${file} ------
+              echo "$RESULT"
+              RETCODE=1
+            fi
+          done
+          exit $RETCODE
+      - name: Whitespace errors
```

Comment:
If I read this correctly, this only applies to `*.cpp` and `*.h` files, although whitespace issues exist for other file types as well (as an aside, we also would like to see newlines at the end of files). Could you update the listing step so Python, MATLAB, etc. files are included in this check?

Reply:
The list of `.cpp` and `.h` files is only used for the Doxygen comment check. The whitespace check is just running on the diff between the PR base and head, and considers any file that Git thinks is a text file.